File: gaia_cron_jobs/config/kafka_config.go
package config

import (
	"fmt"
	"os"
	"strings"

	"github.com/joho/godotenv"
)

type KafkaConfig struct {
	BootstrapServers string
	GroupID          string
	Topics           []string
	Name             string

	ProducerBootstrapServers string
}

func (in *KafkaConfig) LoadEnv() (KafkaConfig, error) {
	err := godotenv.Load(".env")
	if err != nil {
		fmt.Println("Error loading .env file: ", err)
	}

	bootstrapServers := os.Getenv("KAFKA_BOOTSTRAP_SERVERS")
	groupID := os.Getenv("KAFKA_GROUP_ID")
	topics := os.Getenv("KAFKA_TOPICS")
	name := os.Getenv("KAFKA_NAME")
	producerBootstrapServers := os.Getenv("KAFKA_PRODUCER_BOOTSTRAP_SERVERS")

	config := KafkaConfig{
		BootstrapServers:         bootstrapServers,
		GroupID:                  groupID,
		Topics:                   strings.Split(topics, ","),
		Name:                     name,
		ProducerBootstrapServers: producerBootstrapServers,
	}
	return config, nil
}

File: chat_hub/kernel/config/kafka_config.py
import os


class KafkaConfig:
    def __init__(self):
        self.bootstrap_servers = os.getenv('KAFKA_BOOTSTRAP_SERVERS', 'localhost:9092')
        self.group_id = os.getenv('KAFKA_GROUP_ID', 'ai_core')
        self.topics = os.getenv('KAFKA_TOPICS', 'test')
        self.name = os.getenv('KAFKA_NAME', 'ai_core_consumer')
        
def convert_topics_to_list(topics: str = None) -> list:
    if topics :
        return [topic.strip() for topic in topics.split(',')]
    return ['test']
